parsers/NED.py

```diff
@@ -199,10 +202,18 @@ def fetch_production(
     target_datetime = target_datetime or datetime.now(timezone.utc)
 
     json_data = call_api(target_datetime)
-
     NED_data = format_data(json_data, logger)
 
-    return NED_data.to_list()
+    NED_data_list = NED_data.to_list()
+
+    # Add lag to avoid using data that is not yet complete and remove "future" data
+    NED_data_list = [
+        x
+        for x in NED_data_list
+        if x.get("datetime") < (datetime.now(timezone.utc) - timedelta(hours=0.5))
+    ]
```

Comment:
Instead of doing this check here and reconstructing and looping over the entire list can we just check this before we append the event?

Reply:
Yes, I think we could do that. But in this case, should an if statement be added in the case of forecast = True? I'm not sure how the forecasts work, but I suppose we don't want to not append datetimes ahead of now for forecasts?

Reply:
Good point!

I think combining the statements in a single if statement and just adding "`and not forecasts`" would do the trick?

Reply:
Yes, let's do that